portfolio-project-matching-app/models/User.js

```diff
@@ -1,15 +1,85 @@
 class User {
+    // fields
+    id;
+    email;
+    username;
+    introduction;
+    projects;
+    technologies;
 
-    constructor(docSnapshot) {
-        this.id = docSnapshot.id;
-        this.email = docSnapshot.data().email;
-        this.username = docSnapshot.data().username;
-        this.introduction = docSnapshot.data().introduction;
-        this.projects = null;
-        this.technologies = null;
+    // static factory methods (used like overloaded constructors)
+    static fromDocSnapshot(id, docSnapshot) {
```

Comment:
I added some static factory methods in place of having a single constructor. This gives the developer a bit more freedom when creating one of our custom objects / classes.